presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java

```diff
@@ -541,10 +541,6 @@ private static String getColumnName(ConnectorSession session, HiveMetadata metad
     protected boolean isPushdownFilterSupported(ConnectorSession session, TableHandle tableHandle)
     {
         checkArgument(tableHandle.getConnectorHandle() instanceof HiveTableHandle, "pushdownFilter is never supported on a non-hive TableHandle");
-        if (((HiveTableHandle) tableHandle.getConnectorHandle()).getAnalyzePartitionValues().isPresent()) {
-            return false;
-        }
-
```

Comment:
Can you dig a bit on why this was disabled before?

Instead of simply removing the check here, it's best to understand why it's disabled in the first place, I am afraid that there is something behind this decision which can lead to unexpected results if not well understood.

Reply:
FWIW, I did go over the refactoring commit that brought this in (#14040) and I did not find any tests that exercise this that were missed.

My best bet is that this was added a safety valve - in case the call site somehow happens to be from [visitFilter](https://github.com/prestodb/presto/blob/ac3e41e09050fce14190394135c0706ab2a9f067/presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java#L374-L376), which it cannot since the ANALYZE does not have filters